src/UI/Buyer/src/app/product/containers/product-details/product-details.component.ts

```diff
@@ -34,11 +34,22 @@ export class ProductDetailsComponent implements OnInit, AfterViewChecked {
     private appLineItemService: AppLineItemService,
     private appStateService: AppStateService,
     private changeDetectorRef: ChangeDetectorRef,
-    protected favoriteProductService: FavoriteProductsService // used in template
+    protected favoriteProductService: FavoriteProductsService, // used in template
+    private router: Router
   ) {}
 
   ngOnInit(): void {
-    this.getProductData().subscribe((x) => (this.product = x));
+    this.getProductData()
+      .pipe(
+        catchError(() => {
+          return of(null);
```

Comment:
can we add the comment here? I think you can copypasta what's on winmark. It's not really obvious why we're doing what we are

Reply:
Ya totally